### PR TITLE
Redesign Documentation

### LINKED ISSUE #314: Re-order some parts of docs
I think that the `writing tasks` page is way too long right now.

I think it should be considered that some changes be made, like for example badges be moved to API doc.

### PR DESCRIPTION
**Pull request by fkorotkov**
Added top tabs instead of having everything on one page. Check the screenshots:

<details><summary>See Screenshots</summary>

image

image

</details>

Fixes #314

### CODE REVIEW

File: theme/ReadMe.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Mostly copies from https://github.com/squidfunk/mkdocs-material/tree/master/material/partials
```
Comment (RDIL) [inline theme/ReadMe.md]:
```suggestion
Mostly copied from https://github.com/squidfunk/mkdocs-material/tree/master/material/partials
```

File: docs/guide/tips-and-tricks.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  clone_script: |
+    if [[ -z "$CIRRUS_PR" ]]; then
+      git clone --recursive --branch=$CIRRUS_BRANCH https://x-access-token:${CIRRUS_REPO_CLONE_TOKEN}@github.com/${CIRRUS_REPO_FULL_NAME}.git $CIRRUS_WORKING_DIR
+      git reset --hard $CIRRUS_CHANGE_IN_REPO
+    else
+      git clone --recursive https://x-access-token:${CIRRUS_REPO_CLONE_TOKEN}@github.com/${CIRRUS_REPO_FULL_NAME}.git $CIRRUS_WORKING_DIR
+      git fetch origin pull/$CIRRUS_PR/head:pull/$CIRRUS_PR
+      git reset --hard $CIRRUS_CHANGE_IN_REPO
+    fi
+  ...
```
Comment (RDIL) [inline docs/guide/tips-and-tricks.md]:
```suggestion
  # ...
```

File: docs/index.md [hunk 3]
```diff
@@ -1,7 +1,12 @@
-# Cirrus CI
+<h2 class="text-center">Make your development cycle <b>fast</b>, <b>efficient</b>, and <b>secure</b> with modern cloud technologies and Cirrus CI.</h2>
```
Comment (RDIL) [inline docs/index.md]:
```suggestion
<h2 class="text-center">Make your development cycle <b>fast</b>, <b>efficient</b>, and <b>secure</b> with modern cloud technologies through Cirrus CI.</h2>
```

File: docs/index.md [hunk 4]
```diff
@@ -1,7 +1,12 @@
-# Cirrus CI
+<h2 class="text-center">Make your development cycle <b>fast</b>, <b>efficient</b>, and <b>secure</b> with modern cloud technologies and Cirrus CI.</h2> 
 
-Cirrus CI makes your development cycle **fast**, **efficient**, and **secure** by leveraging modern cloud technologies. 
-Cirrus CI scales with your team and makes shipping software faster and cheaper.
+<img src="/assets/images/screenshots/typical-build.png" />
+
+<a href="/features">
+  <button class="btn center">
+    <b>See All The Features</b>
```
Comment (RDIL) [inline docs/index.md]:
```suggestion
    <b>See all the Features</b>
```
typically transition words are lowercase

File: docs/stylesheets/extra.css [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+.center {
+  display: block;
+  margin-left: auto;
+  margin-right: auto;
+}
+
+.text-center {
+  text-align: center;
+}
+
```
Comment (RDIL) [inline docs/stylesheets/extra.css]:
```suggestion
```
extra line at end

File: mkdocs.yml [hunk 6]
```diff
@@ -2,8 +2,9 @@ repo_url: https://github.com/cirruslabs/cirrus-ci-docs/
 site_url: https://cirrus-ci.org
 edit_uri: ""
 
-site_name: Cirrus CI Documentation
+site_name: Cirrus CI
 site_author: Cirrus Labs
+copyright: © Cirrus Labs 2019
```
Comment (RDIL) [inline mkdocs.yml]:
```suggestion
copyright: © Cirrus Labs 2017-present
```
no need to update anymore

File: mkdocs.yml [hunk 7]
```diff
@@ -56,28 +64,34 @@ markdown_extensions:
...
-    - 'Linux Containers': guide/linux.md
-    - 'Windows Containers': guide/windows.md
-    - 'macOS VMs': guide/macOS.md
-    - 'FreeBSD VMs': guide/FreeBSD.md
-    - 'Docker Builder': guide/docker-builder.md
-    - 'Life of a Build': guide/build-life.md
+  - Home:
+    - Home: index.md
+  - Features:
+    - Features: features.md
```
Comment (RDIL) [inline mkdocs.yml]:
```suggestion
    - 'All Features': features.md
```

File: docs/index.md (hunk 4 above)
Comment (fkorotkov) [reply docs/index.md]: I think inside buttons usually every word starts with a capital letter. 🤔

File: mkdocs.yml (hunk 7 above)
Comment (fkorotkov) [reply mkdocs.yml]: This page doesn't contain All Feature but rather features from the previous Key Highlights.

File: docs/index.md (hunk 4 above)
Comment (fkorotkov) [reply docs/index.md]: Changed to `Explore the Features`

**Pull request merged**